### PR TITLE
fix: drop instrumentation from Merlin.make

### PR DESCRIPTION
**Pull request by Alizter**
We drop any instrumentation preprocessing for merlin. This also simplifies some of the code.

fixes https://github.com/ocaml/dune/issues/7465

- [x] changelog
- [x] test

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Unreleased
 ----------
 
+- Fix a bug where merlin would launch the instrumentaiton. (#7606, fixes #7465,
```
Comment (rgrinberg) [inline CHANGES.md:4]: How about:

```
merlin: ignore instrumentation settings for preprocessing
```

**Review by rgrinberg — APPROVED** [review]

**Comment by rgrinberg** [comment]
> I haven't got a test case for this. Not sure how to test.

In the instrumentation tests, you can see how to add setup instrumentation.

In the merlin tests, you can see how to print merlin settings when preprocessors are present.

**Comment by Alizter** [comment]
Oh I still haven't added the tests for this.

**Comment by Alizter** [comment]
Since my ocamlc lives in the nix store, the path is very long, so before PATH_PREFIX_MAP replaces the path, the merlin config gets formatted for the long line. This means that the test is not reproducible depending on the location of ocamlc. I'm not sure if we can fix this since the cram test is doing the substitution.

I guess the simplest solution is just to remove the offending lines. But I am not certain how to do this easily.

**Pull request merged**